drivers/gpio/gpio_stmpe1600.c

```diff
@@ -0,0 +1,297 @@
+
+	LOG_DBG("STMPE1600[0x%02X] init", config->i2c_slave_addr);
+
+	i2c_master = device_get_binding((char *)config->i2c_master_dev_name);
+	if (!i2c_master) {
+		return -EINVAL;
+	}
+	drvdata->i2c_master = i2c_master;
+
+	k_sem_init(&drvdata->lock, 1, 1);
+
+	k_sem_take(&drvdata->lock, K_FOREVER);
+	ret = read_reg16(dev, REG_CHIP_ID_LSB, &chip_id);
+	k_sem_give(&drvdata->lock);
+
+	if (ret) {
+		LOG_ERR("STMPE1600[0x%02X]: Unable to read Chip ID", config->i2c_slave_addr);
+		return ret;
+	}
+
+	if (chip_id != 0x1600) {
+		LOG_ERR("STMPE1600[0x%02X]: Invalid Chip ID", config->i2c_slave_addr);
+		return -EINVAL;
+	}
+
+	return ret;
+}
+
+static int stmpe1600_configure(const struct device *dev,
+			       gpio_pin_t pin, gpio_flags_t flags)
+{
+	struct stmpe1600_drvdata *const drvdata = (struct stmpe1600_drvdata *const) dev->data;
+	const struct stmpe1600_config *const config = dev->config;
+	uint16_t i2c_addr = config->i2c_slave_addr;
+	int ret = 0;
+
+	/* No support for disconnected pin */
+	if ((flags & (GPIO_INPUT | GPIO_OUTPUT)) == GPIO_DISCONNECTED) {
+		return -ENOTSUP;
+	}
+
+	if (k_is_in_isr()) {
+		return -EWOULDBLOCK;
+	}
+
+	k_sem_take(&drvdata->lock, K_FOREVER);
+
+	ret = set_pin_dir(dev, pin, flags);
+	if (ret) {
+		LOG_ERR("STMPE1600[0x%X]: error setting pin direction (%d)", i2c_addr, ret);
+	}
+
+	k_sem_give(&drvdata->lock);
+	return ret;
+}
+
+static int stmpe1600_port_get_raw(const struct device *dev, uint32_t *value)
+{
+	struct stmpe1600_drvdata *const drvdata = (struct stmpe1600_drvdata *const) dev->data;
+	int ret = 0;
+	uint16_t reg_value = 0;
+
+	if (k_is_in_isr()) {
+		return -EWOULDBLOCK;
+	}
+
+	k_sem_take(&drvdata->lock, K_FOREVER);
+
+	ret = read_reg16(dev, REG_GPMR_LSB, &reg_value);
+	if (ret == 0) {
+		*value = reg_value;
+	}
+
+	k_sem_give(&drvdata->lock);
+	return ret;
+}
+
+static int stmpe1600_port_set_masked_raw(const struct device *dev,
+					 uint32_t mask, uint32_t value)
+{
+	struct stmpe1600_drvdata *const drvdata = (struct stmpe1600_drvdata *const)dev->data;
+	uint16_t buf;
+	int ret = 0;
+
+	if (k_is_in_isr()) {
+		return -EWOULDBLOCK;
+	}
+
+	k_sem_take(&drvdata->lock, K_FOREVER);
+
+	if (mask != 0xffff) {
+		ret = read_reg16(dev, REG_GPSR_LSB, &buf);
+	}
+	if (ret == 0) {
+		buf = (buf & ~mask) | (mask & value);
+		ret = write_reg16(dev, REG_GPSR_LSB, buf);
+	}
+
+	k_sem_give(&drvdata->lock);
+	return ret;
+}
+
+static int stmpe1600_port_set_bits_raw(const struct device *dev, uint32_t mask)
+{
+	return stmpe1600_port_set_masked_raw(dev, mask, mask);
+}
+
+static int stmpe1600_port_clear_bits_raw(const struct device *dev, uint32_t mask)
+{
+	return stmpe1600_port_set_masked_raw(dev, mask, 0);
+}
+
+static const struct gpio_driver_api stmpe1600_drv_api = {
+	.pin_configure = stmpe1600_configure,
+	.port_get_raw = stmpe1600_port_get_raw,
+	.port_set_masked_raw = stmpe1600_port_set_masked_raw,
+	.port_set_bits_raw = stmpe1600_port_set_bits_raw,
+	.port_clear_bits_raw = stmpe1600_port_clear_bits_raw,
```

Comment:
You need to add a dummy `pin_interrupt_configure` function that returns `-ENOTSUP`.